aiohttp/web_urldispatcher.py

```diff
@@ -450,17 +453,51 @@ def __init__(self, prefix, directory, *, name=None,
                         'HEAD': ResourceRoute('HEAD', self._handle, self,
                                               expect_handler=expect_handler)}
 
-    def url(self, *, filename, query=None):
-        return str(self.url_for(filename=filename).with_query(query))
+    def url(self, *, filename, append_version=False, query=None):
+        # is there with_query need to be used?
+        # with_query remove previous query options
+        url = self.url_for(filename=filename, append_version=append_version)
+        if query is not None:
+            return str(url.with_query(query))
+        return str(url)
 
-    def url_for(self, *, filename):
+    def url_for(self, *, filename, append_version=False):
         if isinstance(filename, Path):
             filename = str(filename)
         while filename.startswith('/'):
             filename = filename[1:]
         filename = '/' + filename
         url = self._prefix + URL(filename).raw_path
-        return URL(url)
+        url = URL(url)
+        if append_version is True:
+            try:
+                if filename.startswith('/'):
+                    filename = filename[1:]
+                filepath = self._directory.joinpath(filename).resolve()
+                if not self._follow_symlinks:
+                    filepath.relative_to(self._directory)
+            except (ValueError, FileNotFoundError) as error:
+                # relatively safe
+                return url
+            except Exception as error:
+                # perm error or other kind!
+                # TODO log error here?
```

Comment:
Not sure if we should catch the exception.
url is useless: next GET will raise an error.
Moreover next `open` call might fail too.